File: Part2/Fortran/BasicConcepts/BasicConcepts_Redeclaration.f90
program Redeclaration
    implicit none
    
    ! Do not redeclare variable names in most languages:
    character(len=1) :: duplicateCharacter = 'a'
    ! This will cause compile error - variable already declared
    character(len=1) :: duplicateCharacter = 'b'
    
    write(*,*) 'duplicateCharacter:', duplicateCharacter

end program Redeclaration

File: Part2/Go/BasicConcepts/BasicConcepts_Redeclaration.go
package main

import "fmt"

func main() {
	// Do not redeclare variable names in most languages:
	duplicateCharacter := 'a'
	// This will cause compile error - variable already declared
	duplicateCharacter := 'b'

	fmt.Printf("duplicateCharacter: %c\n", duplicateCharacter)
}
